src/chem/aeronet_aod2ioda.py

```diff
@@ -104,76 +97,92 @@ def add_data(infile):
-    var_mdata['sensor_channel'] = writer.FillNcVector(aod_chan, 'integer')
+    for key, value in obsvars.items():
+        varDict[key]['valKey'] = key, iconv.OvalName()
+        varAttrs[key, iconv.OvalName()]['_FillValue'] = -999.
+        varAttrs[key, iconv.OvalName()]['coordinates'] = 'longitude latitude station_elevation'
+        varAttrs[key, iconv.OvalName()]['units'] = 'unitless'
+        varDict[key]['errKey'] = key, iconv.OerrName()
+        varAttrs[key, iconv.OerrName()]['_FillValue'] = -999.
+        varAttrs[key, iconv.OerrName()]['units'] = 'unitless'
+        varAttrs[key, iconv.OerrName()]['coordinates'] = 'longitude latitude station_elevation'
+        varDict[key]['qcKey'] = key, iconv.OqcName()
+        varAttrs[key, iconv.OqcName()]['_FillValue'] = -999
+        varAttrs[key, iconv.OqcName()]['coordinates'] = 'longitude latitude station_elevation'
+        varAttrs[key, iconv.OqcName()]['units'] = ''
 
     for key, value in obsvars.items():
-        outdata[varDict[key]['valKey']] = np.array(f3[value].fillna(nc.default_fillvals['f4']))
-        outdata[varDict[key]['qcKey']] = np.where(outdata[varDict[key]['valKey']] == nc.default_fillvals['f4'],
+        outdata[varDict[key]['valKey']] = np.array(np.float32(f3[value].fillna(np.float32(-999.))))
+        outdata[varDict[key]['qcKey']] = np.where(outdata[varDict[key]['valKey']] == np.float32(-999.),
                                                   1, 0)
-        outdata[varDict[key]['errKey']] = np.where(outdata[varDict[key]['valKey']] == nc.default_fillvals['f4'],
-                                                   nc.default_fillvals['f4'], 0.02)
+        outdata[varDict[key]['errKey']] = np.where(outdata[varDict[key]['valKey']] == np.float32(-999.),
+                                                   np.float32(-999.), np.float32(0.02))
+
+    # Add metadata variables
+    outdata[('latitude', 'MetaData')] = np.array(np.float32(f3['latitude']))
+    outdata[('longitude', 'MetaData')] = np.array(np.float32(f3['longitude']))
+    outdata[('station_elevation', 'MetaData')] = np.array(np.float32(f3['elevation']))
+    varAttrs[('station_elevation', 'MetaData')]['units'] = 'm'
+    outdata[('surface_type', 'MetaData')] = np.full((nlocs), 1)
+    varAttrs[('surface_type', 'MetaData')]['units'] = ''
+
+    c = np.empty([nlocs], dtype=object)
+    c[:] = np.array(f3.siteid)
+    outdata[('station_id', 'MetaData')] = c
+    varAttrs[('station_id', 'MetaData')]['units'] = ''
 
-    # Define global atrributes
-    print('Define global atrributes')
-    AttrData = {'observation_type': 'Aod',
-                'sensor': "aeronet",
-                'surface_type': 'ocean=0,land=1,costal=2'}
+    d = np.empty([nlocs], dtype=object)
+    for i in range(nlocs):
+        d[i] = f3.time[i].strftime('%Y-%m-%dT%H:%M:%SZ')
+    outdata[('datetime', 'MetaData')] = d
+    varAttrs[('datetime', 'MetaData')]['units'] = ''
+
+    outdata[('frequency', 'VarMetaData')] = np.float32(frequency)
```

Comment:
Please place all metadata variables in the group "MetaData". We have deprecated the "VarMetaData" group. Thanks!